R/check_dhis2_attributes-helpers.R

```diff
@@ -1,248 +1,169 @@
+    attribute_id <- paste(attribute_id[idx], collapse = ",")
   }
 
-  list(
-    data_element_group,
-    data_elt_groups
-  )
+  res <- switch(
+    which,
+    "dataSets"          = list(dataset   = attribute_id, # nolint: keyword_quote_linter
+                               data_sets = attributes),
+    "organisationUnits" = list(organisation_unit = attribute_id, # nolint: keyword_quote_linter
+                               org_units         = attributes),
+    "dataElementGroups" = list(data_element_group = attribute_id, # nolint: keyword_quote_linter
+                               data_elt_groups    = attributes),
+    "dataElements"      = attributes # nolint: keyword_quote_linter
+  )
+  res
 }
 
-#' Get the data element identifiers and names
+#' Get the target DHIS2 attribute identifiers and names
 #'
-#' @param username the user name
-#' @param password the user's password
 #' @param base_url the base URL of the DHIS2 server
-#'
-#' @export
-#'
-get_data_elements <- function(base_url, username, password) {
-  checkmate::assertCharacter(base_url, len = 1L, null.ok = TRUE,
-                             any.missing = FALSE)
-  checkmate::assertCharacter(username, len = 1L, null.ok = TRUE,
-                             any.missing = FALSE)
-  checkmate::assertCharacter(password, len = 1L, null.ok = TRUE,
-                             any.missing = FALSE)
-
-  url <- paste0(base_url,
-                "/api/dataElements?fields=id,name,shortName&paging=false")
-  r <- httr::content(httr::GET(url, httr::authenticate(username, password)),
-                     as = "parsed")
-  do.call(rbind.data.frame, r$dataElements)
-}
-
-#' Get the organisation unit identifiers and names
-#'
 #' @param username the user name
 #' @param password the user's password
-#' @param base_url the base URL of the DHIS2 server
+#' @param which the target DHIS2 attribute name.
 #'
+#' @return an object of type `data.frame` with details about the DHIS2
+#'    attributes of interest.
 #' @export
 #'
-get_organisation_units <- function(base_url, username, password) {
-  checkmate::assertCharacter(base_url, len = 1L, null.ok = FALSE,
-                             any.missing = FALSE)
-  checkmate::assertCharacter(username, len = 1L, null.ok = FALSE,
-                             any.missing = FALSE)
-  checkmate::assertCharacter(password, len = 1L, null.ok = FALSE,
-                             any.missing = FALSE)
-
-  url <- paste0(base_url,
-                "/api/organisationUnits?fields=id,name,shortName&paging=false")
-  r <- httr::content(httr::GET(url, httr::authenticate(username, password)),
-                     as = "parsed")
-  do.call(rbind.data.frame, r$organisationUnits)
+#' @examples
+#' \dontrun{
+#' datasets <- get_dhis2_attributes(
+#'   base_url = "https://play.dhis2.org/dev/",
+#'   username = "admin",
+#'   password = "district",
+#'   which = "dataSets"
+#' )
+#' }
+get_dhis2_attributes <- function(base_url = "https://play.dhis2.org/dev/",
```

Comment:
This function has a lot of duplication with the previous one. Can't it be re-used internally?